### ISSUE TITLE
[Feature] Adding cross-dependencies

### ISSUE THREAD

**Issue by thejameskyle** [lerna/lerna#53]
It's kinda awkward right now to `cd` into a package directory and install a dependency, `cd` back up and run `lerna bootstrap` again.

Is there a way we can abstract this?

Something along the lines of:

``` sh
$ lerna add [package] dependency
$ lerna add [package] dependency --dev
$ lerna add [package] dependency --peer
```

Update: We should match Yarn's API for adding.

**Comment by ianwremmel** [comment]
I think I see two different needs that could be addressed in almost the same way:

1. Add a dependency to a single package like you described (but maybe closer to standard npm commands):

  ```
  $ lerna install [--save] my-package-name dep-name [dep-name-two...]
  ```
2. Add a dependency to all packages:

  ```
  $ lerna install [--save] dep-name [dep-name-two...]
  ```
or possibly
  ```
  $ lerna install [--save] * dep-name [dep-name-two...]
  ```

**Comment by thejameskyle** [comment]
Fixed in 2.0.0-beta.4

**Comment by rygine** [comment]
the `install` command could take advantage of the `scope` flag. the `ignore` flag may land soon, which could be used as well.

if you're currently in a package folder, it would make sense that `lerna install` would add something to that package, unless otherwise specified.

**Comment by gigabo** [comment]
Asini [has this](https://github.com/asini/asini/pull/46).  We should port that over.

**Comment by reggi** [comment]
What's the progress with this? How do people currently do this? Manually enter it in the `package.json`?

**Comment by kirbysayshi** [comment]
One way I worked around this for now:

```
$ lerna exec -- npm install --save debug@latest
```

**Comment by spudly** [comment]
I was getting by with just running `npm install--save <package>` before, but now that I've enabled automatic hoisting I want it to install the package at the root so normal install won't work.

Current workflow:
1. Run `npm show <pkg> version` to determine what version to use
2. Manually add the dependency to `package.json`
3. Run `lerna bootstrap --scope="<my pkg>"`

Adding this feature would save me a lot of time!

**Comment by jribeiro** [comment]
Hi @thejameskyle, regarding https://github.com/lerna/lerna/issues/53#issue-133325098, do you know if this is still planned to be implemented in `lerna` or on the experimental `yarn` workspaces feature? Thanks!

**Comment by TxHawks** [comment]
Is there a PR for this? Is there interest in one?

**Comment by marionebl** [comment]
Implemented via #1069 and released via https://github.com/lerna/lerna/releases/tag/v2.5.0
